psi4/driver/molutil.py

```diff
@@ -209,70 +58,115 @@ def molecule_get_attr(self, name):
-            Black.append(u)
-            Fragment[-1].append(int(u))  # add to group (0-indexed)
-            Fragment[-1].sort()          # preserve original atom ordering
-
-        if White:                        # can't move White -> Queue if empty
-            Queue.append(White[0])
-            White.remove(White[0])
-
-    return Fragment
+@classmethod
+def molecule_from_string(cls,
+                         molstr,
+                         dtype=None,
+                         name=None,
+                         fix_com=None,
+                         fix_orientation=None,
+                         fix_symmetry=None,
+                         return_dict=False,
+                         enable_qm=True,
+                         enable_efp=True,
+                         missing_enabled_return_qm='none',
+                         missing_enabled_return_efp='none',
+                         verbose=1):
+    molrec = qcdb.molparse.from_string(molstr=molstr,
+                                       dtype=dtype,
+                                       name=name,
+                                       fix_com=fix_com,
+                                       fix_orientation=fix_orientation,
+                                       fix_symmetry=fix_symmetry,
+                                       return_processed=False,
+                                       enable_qm=enable_qm,
+                                       enable_efp=enable_efp,
+                                       missing_enabled_return_qm=missing_enabled_return_qm,
+                                       missing_enabled_return_efp=missing_enabled_return_efp,
+                                       verbose=verbose)
+    if return_dict:
+        return core.Molecule.from_dict(molrec['qm']), molrec
+    else:
+        return core.Molecule.from_dict(molrec['qm'])
+
+
+@classmethod
+def molecule_from_arrays(cls,
+                         geom=None,
+                         elea=None,
+                         elez=None,
+                         elem=None,
+                         mass=None,
+                         real=None,
+                         elbl=None,
+
+                         name=None,
+                         units='Angstrom',
+                         input_units_to_au=None,
+                         fix_com=None,
+                         fix_orientation=None,
+                         fix_symmetry=None,
+
+                         fragment_separators=None,
+                         fragment_charges=None,
+                         fragment_multiplicities=None,
+
+                         molecular_charge=None,
+                         molecular_multiplicity=None,
+
+                         missing_enabled_return='error',
+                         tooclose=0.1,
+                         zero_ghost_fragments=False,
+                         nonphysical=False,
+                         mtol=1.e-3,
+                         verbose=1,
+
```

Comment:
This is quite a few new arrays. Anything that needs to go back to the QC Schema?

Reply:
except for `name`, they're all in there. the last block (missing_ ... mtol) are controlling _how_ it's processed, so not for the schema.